alf/algorithms/rl_algorithm.py

```diff
@@ -204,12 +258,23 @@ def __init__(self,
         self._current_transform_state = None
 
         if self._env is not None and not self.on_policy:
-            if config.whole_replay_buffer_training and config.clear_replay_buffer:
-                replayer = "one_time"
-            else:
-                replayer = "uniform"
-            self.set_exp_replayer(replayer, self._env.batch_size,
-                                  config.replay_buffer_length,
+            replay_buffer_length = adjust_replay_buffer_length(
+                config, self._num_earliest_frames_ignored)
+
+            if config.whole_replay_buffer_training:
+                # For whole replay buffer training, we would like to be sure
+                # that the replay buffer have enough samples in it to perform
+                # the training, which will most likely happen in the 2nd
+                # iteration. The minimum_initial_collect_steps guarantees that.
+                minimum_initial_collect_steps = replay_buffer_length * self._env.batch_size
```

Comment:
should this be `(unroll_length + num_earliest_frames_ignored) * self._env.batch_size` ?

Reply:
Here the `replay_buffer_length` is already adjusted, which should have included `num_earliest_frames_ignored` if it is set.

Reply:
replay_buffer_length can be very big if clear_replay_buffer is False.